invisible_cities/cities/components_test.py

```diff
@@ -12,7 +12,6 @@
 
```

Comment:
This test was supposed to fail if the implementation of `compute_xy_position` doesn't take into account the correct run number. With your modification what fails is the code itself, since an exception is raised. My understanding is that in the test we want a test to fail, not the code itself, to make it clear what is failing; in fact, the code could be failing righteously, tell me if I'm wrong.

Reply:
> This test was supposed to fail if the implementation of compute_xy_position doesn't take into account the correct run number

And the result of not taking into account the correct run number is that it raises a `ClusterEmptyList` exception, which is a valid failure for a test. 

> With your modification what fails is the code itself, since an exception is raised. My understanding is that in the test we want a test to fail, not the code itself.

What do you mean that what fails is the code? A test passes if and only if no exception is raised, otherwise it fails.  Most tests contain `assert`, which raises an `AssertionError` exception if a condition is not met, but `assert` is just a convenience for something like
```
if not <condition>:
    raise AssertionError
```

> in fact, the code could be failing with good reason

What do you mean? If the code fails for a different reason the test will still fail (in both cases)

Reply:
Yes, I understand that logically it is exactly the same, I just thought it was clearer to have an assertion in the test, but it doesn't matter.